connector-framework/src/main/java/org/wso2/carbon/connector/framework/ConnectorManager.java

```diff
@@ -105,8 +112,18 @@ public ClientConnector getClientConnector(String protocol) {
         return clientConnectors.get(protocol);
     }
 
-    private void registerServerConnectorProvider(ServerConnectorProvider serverConnectorProvider) {
+    /**
+     * Register ServerConnectorProvider to the ConnectionManager.
+     * @param serverConnectorProvider to be register
+     */
+    public void registerServerConnectorProvider(ServerConnectorProvider serverConnectorProvider) {
         serverConnectorProviders.put(serverConnectorProvider.getProtocol(), serverConnectorProvider);
+        if (DataHolder.getInstance().getBundleContext() != null) {
```

Comment:
This will not work in a non-OSGi env as we are strictly checking for bundleContext availability and then calling initializeConnectors.